mixer/adapter/fluentd/fluentd_test.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fluentd
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/fluent/fluent-logger-golang/fluent"
+
+	descriptor "istio.io/api/mixer/v1/config/descriptor"
+	"istio.io/istio/mixer/adapter/fluentd/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/adapter/test"
+	"istio.io/istio/mixer/template/logentry"
+)
+
+func TestBasic(t *testing.T) {
+	info := GetInfo()
+
+	if !contains(info.SupportedTemplates, logentry.TemplateName) {
+		t.Error("Didn't find all expected supported templates")
+	}
+
+	cfg := info.DefaultConfig
+	b := info.NewBuilder().(*builder)
+	b.SetAdapterConfig(cfg)
+
+	if err := b.Validate(); err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+
+	handler, err := b.injectBuild(context.Background(), test.NewEnv(t), &fluent.Fluent{})
+	if err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+
+	logEntryHandler := handler.(logentry.Handler)
+	err = logEntryHandler.HandleLogEntry(context.Background(), nil)
+	if err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+
+	if err = handler.Close(); err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+}
+
+func contains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func TestBuilder(t *testing.T) {
+	env := test.NewEnv(t)
+
+	cases := []struct {
+		config  config.Params
+		success bool
+		inject  bool
+	}{
+		{config.Params{
+			Address: "",
+		}, false, false},
+		{config.Params{
+			Address: "dummy",
+		}, false, false},
+		{config.Params{
+			Address: "1.2.3.4:1234",
+		}, true, true},
+	}
+
+	for i, c := range cases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
```

Comment:
why not just name the tests for the Address being provided?  then, upon failure, I know what the input was precisely.

Reply:
Sounds good, let me figure this out.

Reply:
Done